### PR TITLE
Spectroscopy QoI example

### PR DESCRIPTION
**Pull request by tradowsk**
A new example for the Spectroscopy QoI.

SpectroscopicAbsorption QoI only calculates absorption value at a single wavenumber. This example runs the QoI in a loop to produce a plot of absorption vs wavenumber for a given range specified in the input file.

`SpectroscopyExample/wavenumber_range` specifies `'nu_min nu_max nu_step'` and `SpectroscopyExample/output_prefix` specifies the prefix for the output file where the data is written.

### CODE REVIEW

**Comment by pbauman** [comment]
How about instead of a proper "app", we just make this an example in the examples directory? This seems awfully specialized to ship as an app, but having as an example illustrates building a user app around this capability. Let me know if you need help with the automake foo in examples.

File: src/qoi/include/grins/absorption_coeff.h [hunk 1]
```diff
@@ -78,6 +78,11 @@ namespace GRINS
                     libMesh::Real desired_nu, const std::string & species,
                     libMesh::Real thermo_pressure);
 
+    void set_desired_wavenumber(libMesh::Real new_nu)
```
Comment (pbauman) [inline src/qoi/include/grins/absorption_coeff.h]: `set_wavenumber` is clear enough I think.

**Comment by tradowsk** [comment]
Okay, I have converted the app to an example.

Example input file uses a 10x10 mesh and has homogeneous conditions for simplicity. Also includes all the necessary spectroscopy data files to scan the wavenumber range [2717.0,2174.0] cm^-1 with a step of 0.01 cm^-1. Run takes about 5 seconds total on my machine.

**Comment by pbauman** [comment]
Please rebase on master since #536 has been merged. Also, the nooptdeps test is failing to build, looks like you're not guarding the code in the new example if there's no Antioch.

**Comment by tradowsk** [comment]
I always forget about guarding for Antioch, sorry...

Would the best way be to guard the body of the example `main()` function and add a `libmesh_error_msg` when not build with Antioch?

**Comment by pbauman** [comment]
> Would the best way be to guard the body of the example main() function and add a libmesh_error_msg when not build with Antioch?

That's fine with me. Just make sure you indicate that Antioch's what missing and GRINS should be configured with Antioch for that example to work.

**Comment by pbauman** [comment]
This is ready to go?

**Comment by tradowsk** [comment]
Yes, was just waiting for femputer to finish (was really slow for some reason)

**Comment by roystgnr** [comment]
> Yes, was just waiting for femputer to finish (was really slow for some reason)

:sweat_smile:

**Pull request merged**